src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,342 @@
+      <FormControl
+        label={props.label}
+        labelId={ids.label}
+        id={ids.textbox}
+        error={validationMessage}
+        onKeyDown={onKeyDown}
+      >
+        <div role="presentation" className={classNames.formControlChildrenContainer} onClick={onClick}>
+          <TagList
+            className={classNames.tagList}
+            labelledBy={ids.label}
+            refs={valueRefs}
+          >
+            {value.length !== 0 &&
+              props.tagChildren ?
+              props.tagChildren(value, valueRefs, onOptionRemove) :
+              value.map((val, index) => (
+                <Tag
+                  defaultActive={index === 0}
+                  id={`${props.id}-option-${props.optionIDGetter(getOption(val))}`}
+                  key={props.optionIDGetter(getOption(val))}
+                  onRemove={onOptionRemove}
+                  readOnly={props.readOnly}
+                  ref={valueRefs[index]}
+                >
+                  {props.optionLabelGetter(
+                    props.options.find(option => props.optionIDGetter(option) === props.optionIDGetter(getOption(val))),
+                  )}
+                </Tag>
+              ))
+            }
+            <input
+              aria-autocomplete="list"
+              aria-controls={ids.listbox}
+              aria-describedby={`${ids.emptyMessage}`}
+              aria-expanded={expanded}
+              aria-haspopup="listbox"
+              autoComplete="off"
+              role="combobox"
+              className={classNames.input}
+              id={ids.textbox}
+              onBlur={onAutocompleteBlur}
+              onChange={onAutocompleteChange}
+              onInput={props.onInput}
+              placeholder={value.length === 0 ? props.placeholder : undefined}
+              readOnly={props.readOnly}
+              required={props.required ? value.length === 0 : false}
+              ref={autocompleteRef}
+              value={autocompleteValue}
+            />
+          </TagList>
+          <FormControlIcons validationMessage={validationMessage} className={classNames.iconsContainer}>
+            {(!props.readOnly && value.length > 0) && (
+              <IconButton
+                icon={iconClear}
+                label={`Remove all selected options on ${props.label}`}
+                onPress={onOptionsClear}
+                className={classNames.iconClear}
+              />
+            )}
+            {props.readOnly ? (
+              <Icon icon={iconCaretDown} label={`Opening ${props.label} options disabled while read only`} />
+            ) : (
+              <IconButton
+                icon={iconCaretDown}
+                label={`Open ${props.label} options`}
+                onPress={onClick}
+              />
+            )}
+          </FormControlIcons>
+        </div>
+      </FormControl>
+      {dropdownContents()}
+    </div>
+  );
+});
+
+MultiSelect.propTypes = {
```

Comment:
I realized if we're going to be okay with missing test coverage on "pass through props", there's no point in double check the test coverage for our props interfaces? I kinda seems silly now.

Reply:
Yeah, this test suite was written before our latest conversation on the subject, so it does indeed have full coverage. 🙃